### PR TITLE
Make crash reporting optional

### LINKED ISSUE #2743: Make crash reporting optional
### Is your feature request related to a problem? Please describe.
Starting February 2022 users can see which type of information apps collect and if it's optional or not: https://android-developers.googleblog.com/2021/10/launching-data-safety-in-play-console.html
For openHAB "Crash logs and Diagnostics" won't be marked as optional.

### Describe the solution you'd like
Add a setting to disable crash reporting.

### PR DESCRIPTION
**Pull request by wiomoc**
Closes #2743

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt [hunk 1]
```diff
@@ -396,6 +398,15 @@ class PreferencesActivity : AbstractBaseActivity() {
                 preferenceScreen.removePreferenceRecursively(PrefKeys.TASKER_PLUGIN_ENABLED)
             }
 
+            if (Util.isFlavorFoss) {
+                preferenceScreen.removePreference(crashReporting)
+            } else {
+                crashReporting.setOnPreferenceClickListener {
+                    CrashReportingHelper.initialize(requireActivity().application)
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt:405]: AFAIK this is called before the changed value is saved.

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt [hunk 2]
```diff
@@ -396,6 +398,15 @@ class PreferencesActivity : AbstractBaseActivity() {
                 preferenceScreen.removePreferenceRecursively(PrefKeys.TASKER_PLUGIN_ENABLED)
             }
 
+            if (Util.isFlavorFoss) {
+                preferenceScreen.removePreference(crashReporting)
+            } else {
+                crashReporting.setOnPreferenceClickListener {
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt:404]: `setOnPreferenceChangeListener`

**Review by mueller-ma — CHANGES_REQUESTED** [review]

File: mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt (hunk 1 above)
Comment (wiomoc) [reply mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt:405]: It's actually the other way around; `setOnPreferenceChangeListener` is called before the change is committed, `setOnPreferenceClickListener` after.

**Review by mueller-ma — APPROVED** [review]
LGTM!

**Pull request merged**